### ISSUE TITLE
Accent color for title bars and window borders

### ISSUE THREAD

**Issue by Belim** [Kinnara/ModernWpf#171]
Is there a possibility to inherit these settings?

![ApplicationFrameHost_riDvCms0r4](https://user-images.githubusercontent.com/57478606/93025428-246f6580-f5fe-11ea-95a6-c12c25142fdb.png)

**Comment by ShankarBUS** [comment]
I guess there should be.

Try overriding the `WindowBorder` resource with a brush of your interest.

For example :
```xaml
<Window.Resources>
    <SolidColorBrush x:Key="WindowBorder" Color="Purple"/>
</Window.Resources>

You can override this resource not only from the window but from App.xaml too (it will be applied to all windows inside your application).

**Comment by Belim** [comment]
Hi ShankarBUS,
i realize already, it is so damn easy if you know how to do it. Next time i try to be a little more ambitious :grin:

Many thanks for the speedy responses  :blush:

**Comment by ShankarBUS** [comment]
> Many thanks for the speedy responses

I'm always online and I'm a fan of this library. So I rush to close as many issues possible. I want to compete with the author of this library on fixing issues 😅.

**Comment by Belim** [comment]
and I thought you were the developer :grin:

another addition. the option `x:Key="WindowBorder"` does not inherit the Windows settings and is always active if set. is this a bug or am I doing something wrong?

`Titlebar.*` works like a charm

**Comment by ShankarBUS** [comment]
> another addition. the option x:Key="WindowBorder" does not inherit the Windows settings and is always active if set. is this a bug or am I doing something wrong?

Can you explain a little more clearly? I don't understand.

So the window shows the same border even after the window is deactivated, Is it what you're saying?

But it works for me. The windows shows a different border for inactive state.

If you want to set a border for inactive state you can override `WindowBorderInactive`

**Comment by Belim** [comment]
a picture is worth a thousand words
![issue](https://user-images.githubusercontent.com/57478606/93055430-33403180-f66b-11ea-8324-08b5fa6619ac.png)

> If you want to set a border for inactive state you can override WindowBorderInactive

Maybe this could be the solution

**Comment by ShankarBUS** [comment]
Ohhhh! Ok now I got it.

The border is set by you **manually** you can't expect it to work.
This library doesn't support coloured window borders so I told you a method to change the border. That won't follow Windows Settings.

You have to do the work for that.

**Comment by ShankarBUS** [comment]
So UWP apps too don't respect the settting either
![image](https://user-images.githubusercontent.com/63704247/93056573-3c8cc680-f68a-11ea-8775-1d623c9d5992.png)

**Comment by ShankarBUS** [comment]
So the ControlUWP app is a .NET Framework 4.8 app. Which is a good thing because you use registry to fetch the said setting for coloured titlebar/window borders.

**Comment by ShankarBUS** [comment]
Hmmm, this is a valid request maybe you should open a feature request for this

**Comment by Belim** [comment]
Reopening the issue should do the trick ?:smiley:

**Comment by ShankarBUS** [comment]
Hey @Belim, I will provide a sample code.

> Reopening the issue should do the trick ?😃

Yes, you can also edit this to include your needs

**Comment by ShankarBUS** [comment]
Hey @Belim,

Until the bug is fixed you can do this :

```c
using System;
using System.Diagnostics;
using System.Management;
using System.Security.Principal;

namespace Test
{
    public class WindowBorderHelper
    {
        private bool _init = false;
        private const string DWMRegKey = "SOFTWARE\\Microsoft\\Windows\\DWM";
        private const string ColorValueName = "ColorPrevalence";

        public static event EventHandler SystemThemeChanged;

        public static bool GetUseColoredWindowBorders() => ReadDWord(ColorValueName);

        private static bool ReadDWord(string valueName)
        {
            var regkey = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(DWMRegKey);
            if (regkey == null)
            {
                return false;
            }

            return (int)regkey.GetValue(valueName, 0) > 0;
        }

        public static void Initialize()
        {
            if (_init = false)
            {
                _init = true;
            }
            else
            {
                return;
            }
            
            Update();
            try
            {
                var currentUser = WindowsIdentity.GetCurrent();
                var query = new WqlEventQuery(string.Format("SELECT * FROM RegistryValueChangeEvent WHERE Hive='HKEY_USERS' AND KeyPath='{0}\\\\{1}' AND ValueName='{2}'",
                currentUser.User.Value, DWMRegKey.Replace("\\", "\\\\"), ColorValueName));

                ManagementEventWatcher watcher = new ManagementEventWatcher(query);

                watcher.EventArrived += new EventArrivedEventHandler(HandleEvent);

                watcher.Start();
            }
            catch (ManagementException managementException)
            {
                Debug.WriteLine(managementException.Message);
            }
        }

        private static void HandleEvent(object sender, EventArrivedEventArgs e)
        {
            Update();
        }

        private static void Update()
        {
            SystemThemeChanged?.Invoke(null, null);
        }
    }
}
```

You can use this to fetch the system settings for the coloured window chromes. Place this following code on window loaded method (or anywhere you want)

```cs
WindowBorderHelper.SystemThemeChanged += (_, __) =>
{
    this.BorderBrush = WindowBorderHelper.GetUseColoredWindowBorders() ? Brushes.Purple : Brushes.Black;
};

WindowBorderHelper.Initialize();
```
This is a ugly workaround, it should be supported by this library itself. You can create your method of updating the window. But the basic concept is the registry and listening to its changes.

Caution : I Didn't use visual studio to create these codes just used notepad to type it. You may expect some issues.

**Comment by Belim** [comment]
Many thanks for the code snippet. 
I can accept that currently. If i add your workaround, a new pull request for ControlUWP would be appropriate (since your code) :relaxed:

**Comment by ShankarBUS** [comment]
Did you try it? Does it work?

> If i add your workaround, a new pull request for ControlUWP would be appropriate

Thanks for the offer but this is a really small feature to make a separate PR. I currently don't have my PC so I wouldn't be able to make it.

> (since your code) ☺

This is GitHub it is **"OUR"** code now.

This workaround could be added to ModernWpf so that other users can be benefited.

If you made it work with your app you can make a PR in this library.

**Comment by Kinnara** [comment]
I recall encountering some complexities when trying to implement this, so the feature didn't make it to the first version (and consequently, any version). Now we have the motivation to take a second look.

**Comment by ShankarBUS** [comment]
@Kinnara

Yes having it in the base UI library will reduce overhead work for devs.

But how are you going to implement this?

If you did the same as I did you will end up with adding `Microsoft.Win32.Registry`, `System.Management` & some other dependencies for .NET Core target.

@Belim's app was a .NET Framework app so it won't be a problem for him. But if a .NET Core app used this library they will end up with additional dependencies which are unnecessary for them (we already have to take a dependency on `Microsoft.Windows.SDK.NET` which is a 25 MB library for .NET 5 target. New problems arise in distribution of apps due to heavy packages, runtime & dependencies)

Are there any WinRT API for this? So that we reduce the package deps?

**Comment by Belim** [comment]
> I recall encountering some complexities when trying to implement this, so the feature didn't make it to the first version (and consequently, any version). Now we have the motivation to take a second look.

Buy you a beer'?  (as motivation) 😁

**Comment by ShankarBUS** [comment]
> Buy you a beer'? (as motivation) 😁

🤣🤣🤣

**Comment by Belim** [comment]
> > Buy you a beer'? (as motivation) 😁
> 
> 🤣🤣🤣

You'll get a wodka :smirk: let's go to work :sleeping: :grin:

**Comment by ShankarBUS** [comment]
@Belim 🍺🍻
